### ISSUE TITLE
Delivery Status Notification is not always sent

### ISSUE THREAD

**Issue by Salo15** [sympa-community/sympa#852]
Sympa does not always send a Delivery Status Notification (DSN) if a sender does not have write permission for a list and according to the scenario a DSN should actually be sent (with the information that the authorization was denied).

After raising the log level and restarting Sympa, the DSN is sent correctly.

Version
-------
6.2.16

Installation method
-------------------
Debian Package

Expected behavior
-----------------
For testing I use a test list that uses the following standard send scenario:

[root@lists4(KIT) ~]# more /usr/share/sympa/default/scenari/send.newsletter
title.gettext Newsletter, restricted to moderators

is_editor([listname],[sender]) smtp,dkim,smime,md5 -> do_it
true() smtp,dkim,smime,md5 -> reject(reason='send_editor')

If I send an email to this list from an address that is not registered as a moderator of the list, a Delivery Status Notification should be sent.

Actual behavior
---------------
If I send an email to this list from an address that is not registered as a moderator of the list, the email will be rejected, but no DSN will be sent.
I can see in the Sympa log that the email is rejected but there is no entry for sending an email to the sender, i.e. the email is actually not sent and I can rule out that this is the problem that the email ends up in a spam filter.

Strangely enough, however, I have made the following observation:

After raising the log level in the Sympa configuration (and restarting Sympa) the DSN is sent correctly.

In this case the DSN arrives and I see the following line in the Sympa log, which indicates that this mail was sent:
Jan 27 11:06:07 lists4.scc.kit.edu sympa_msg[9390]: notice Sympa::Spindle::ProcessTemplate::_twist() Processing Sympa::Message::Template <sabinetest3-l@lists.kit.edu.1580119567.9390,145/shelved:smime_sign>; envelope_sender=<>; message_id=sympa.1580119567.9390.236@lists.kit.edu; recipients=sabine.lorenz@kit.edu; sender=sympa-request@lists.kit.edu; template=delivery_status_notification; action=failed; reason=send_editor; status=5.7.1

One minute before (before the log level was raised) I had sent a test mail from the same sender to the same list address (not changing the list config or anything else) and the DSN had not been sent.

Additional information
----------------------
In addition to the mailing list server of KIT we also operate the mailing list server for DFN (with the same Sympa version).
On this server, too, I have noticed the identical behavior with a test list.

**Comment by ikedas** [comment]
Perhaps this issue is a duplicate of issue #820 .

**Comment by Salo15** [comment]
I think I can confirm that. 
I have entered these lines to delete the "Quiet" flag on our server in .../lib/Sympa/Spindle/ProcessIncoming.pm and restarted Sympa. Since then, the test mail is sent in all cases (even after I have performed an action with "quiet" once, which in my case before this change actually resulted in no rejection report being sent for a list).

Thank you very much and best regards,
Sabine
